presto-spi/src/main/java/com/facebook/presto/spi/type/Chars.java

```diff
@@ -95,6 +105,36 @@ private static int compareCharsShorterToLonger(Slice shorter, Slice longer)
         return 0;
     }
 
+    private static int compareCharsShorterToLongerNoPad(Slice shorter, Slice longer, int shorterLength, int longerLength)
+    {
+        for (int i = 0; i < shorterLength; ++i) {
+            int result;
+            if (i >= shorter.length()) {
+                if (i >= longer.length()) {
+                    return 0;
+                }
+                result = compareUnsignedBytes((byte) ' ', longer.getByte(i));
+            }
+            else {
+                if (i >= longer.length()) {
+                    if (i >= shorter.length()) {
```

Comment:
No, it happens e.g.:
CHAR(2) = '  ' vs CHAR(1) = ' '
then:
longerLength = 2
shorterLength = 1
longer.length() = 0
shorter.length() = 0